py3status/modules/file_status.py

```diff
@@ -57,32 +88,73 @@ class Meta:
         }
 
     def post_config_hook(self):
-        if self.path:
-            self.path = expanduser(self.path)
+        if not self.path:
+            raise Exception(STRING_NO_PATH)
+
+        # deprecation
+        on = getattr(self, 'icon_available', None)
+        off = getattr(self, 'icon_unavailable', None)
+        if self.format == DEFAULT_FORMAT and (on or off):
+            self.format = u'\?if=paths {}|{}'.format(on or u'●', off or u'■')
+            new_format = u'\?color=paths [\?if=paths {}|{}]'
```

Comment:
We forget to replace `{icon}` with `new_format`. See if you can fix this.

Reply:
Not sure to understand, we don't need ~~{icon}~~ in new format no? And if I use legacy style I see my ~~icon_available~~ or ~~icon_unavailable~~

Reply:
That removes the `icon_*` okay... and for default format only, but not the old existing format.
We need to keep the old format alive too... eg `format = 'VPN {icon}`.